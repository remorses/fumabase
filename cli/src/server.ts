import { WebSocketServer } from 'ws'
import { startTunnel } from 'untun'
import net from 'net'

/**
 * Finds an available port on localhost and returns it.
 * @returns {Promise<number>} The available port found.
 */
async function findAvailablePort(): Promise<number> {
    return new Promise((resolve, reject) => {
        const server = net.createServer()
        server.listen(0, () => {
            const address = server.address()
            if (typeof address === 'object' && address?.port) {
                const port = address.port
                server.close(() => resolve(port))
            } else {
                server.close(() => reject(new Error('failed to get port')))
            }
        })
        server.on('error', reject)
    })
}
type WebSocketWithTunnel = {
    wss: WebSocketServer
    tunnel: any
    port: number
    stop: () => void
    websocketUrl: string
}

export function startWebSocketWithTunnel({ port }: { port?: number } = {}) {
    return new Promise<WebSocketWithTunnel>(async (resolve, reject) => {
        let actualPort: number
        try {
            actualPort = port ?? (await findAvailablePort())
        } catch (err) {
            reject(err)
            return
        }

        // 1) local WebSocket server
        const wss = new WebSocketServer({ port: actualPort }, () =>
            console.log(`ws://localhost:${actualPort} ready`),
        )

        // 2) untun tunnel
        const start = Date.now()

        try {
            const tunnel = await startTunnel({
                port: actualPort,
                acceptCloudflareNotice: true,
                hostname: 'localhost',
                protocol: 'http',
            })
            if (!tunnel) throw new Error('Failed to create Cloudflare tunnel')

            const websocketUrl = (await tunnel.getURL()).replace(
                /^https?:/,
                'wss:',
            )
            const elapsed = Date.now() - start
            console.log(`🌐 Public WebSocket endpoint → ${websocketUrl}`)
            console.log(
                `⏱ Tunnel URL obtained in ${(elapsed / 1000).toFixed(2)}s`,
            )

            // 3) graceful shutdown
            const stop = () => {
                console.log('\n⏹ shutting down…')
                if (
                    tunnel &&
                    'close' in tunnel &&
                    typeof tunnel.close === 'function'
                ) {
                    tunnel.close()
                }
                wss.close()
            }

            const shutdown = () => {
                stop()
                process.exit()
            }

            process.on('SIGINT', shutdown)
            process.on('SIGTERM', shutdown)

            resolve({
                wss,
                tunnel,
                port: actualPort,
                websocketUrl,
                stop,
            })
        } catch (err) {
            reject(err)
        }
    })
}
