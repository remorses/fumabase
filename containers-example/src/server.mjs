import * as http from 'http';

const server = http.createServer((req, res) => {
  res.writeHead(200, { 'Content-Type': 'text/plain' });
  res.end('hello\n');
});

server.listen(8080, () => {
  console.log('Server running at http://localhost:8080/');
});
