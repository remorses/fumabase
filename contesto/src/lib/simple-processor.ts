import { Root } from 'mdast'

import { remark } from 'remark'
import { Processor } from 'unified'
import remarkGfm from 'remark-gfm'
import { memoize } from './utils.js'

export const simplerProcessor = remark().use(remarkGfm)

type ProcessorData = { ast: any }

export const processorWithAst = memoize(
    function processorWithAst(processor): Processor {
        return processor().use(() => (tree, file) => {
            if (!file.data) file.data = {}
            const data: ProcessorData = file.data
            if (tree) {
                data.ast = tree
            }
        })
    },
    { maxSize: 10 },
)
