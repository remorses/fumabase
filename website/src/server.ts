import { createHonoServer } from 'react-router-hono-server/node'

export default await createHonoServer({
    port: Number(process.env.PORT) || 7664,
}) as any
