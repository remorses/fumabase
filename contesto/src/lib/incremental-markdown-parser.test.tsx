import { describe, expect, test } from 'vitest'
import { SafeMdxRenderer } from 'safe-mdx'
import { renderToStaticMarkup } from 'react-dom/server'
import { simplerProcessor } from './simple-processor.js'

import {
    IncrementalParsingProps,
    parseMarkdownIncremental,
    SegmentEntry,
} from './incremental-markdown-parser.js'
import React from 'react'

// Wait for highlighter to initialize
async function parseAndWaitForHighlighter(props: IncrementalParsingProps) {
    let retries = 0
    while (retries < 10) {
        try {
            return parseMarkdownIncremental({
                ...props,
            })
        } catch (promise) {
            if (promise instanceof Promise) {
                await promise
                retries++
            } else {
                throw promise
            }
        }
    }
    throw new Error('Highlighter initialization timed out')
}

function toMarkdown(ast: any): string {
    const file = simplerProcessor.stringify(ast)
    return String(file)
}

function simulateStreaming(text: string): string[] {
    const words = text.split(' ')
    const result: string[] = []
    for (let i = 1; i <= words.length; i++) {
        result.push(words.slice(0, i).join(' '))
    }
    return result
}

describe('incremental parse', () => {
    const cache = new Map<number, SegmentEntry>()
    const extension = 'mdx'
    const markdown = `
# Welcome to the Example Document

This is a sample markdown file to showcase various features.

## Features

* Easy formatting
* Lists
* **Bold text**
* [Links](https://example.com)

> "Markdown makes documentation easy!"

\`\`\`js
console.log('Hello, world!');
\`\`\`

1. one
2. two
3. three

this is another paragraph

<Cards>
  <Card>
  some text
  </Card>
</Cards>


`.trim()

    // Loop over streaming chunks and parse incrementally
    const streamingChunks = simulateStreaming(markdown)
    streamingChunks.forEach((chunk, idx) => {
        test(`incremental parsing at chunk ${idx + 1}`, async () => {
            try {
                const ast = await parseAndWaitForHighlighter({
                    markdown: chunk,
                    cache,
                })
                const md = toMarkdown(ast)
                // const changes = diffWords(chunk, md)
                // const additions = changes.filter((change) => change.added)
                expect(md).toMatchSnapshot()
            } catch (error) {
                expect(error.message).toMatchSnapshot()
            }
        })
        test(`incremental parsing and jsx at chunk ${idx + 1}`, async () => {
            try {
                const ast = await parseAndWaitForHighlighter({
                    markdown: chunk,
                    cache,
                })
                const jsx = renderToStaticMarkup(
                    <SafeMdxRenderer

                        mdast={ast}
                        components={{
                            Cards({ children }) {
                                return React.createElement(
                                    'cards',
                                    null,
                                    children,
                                )
                            },
                            Card({ children }) {
                                return React.createElement(
                                    'card',
                                    null,
                                    children,
                                )
                            },
                        }}
                    />,
                )
                // const changes = diffWords(chunk, md)
                // const additions = changes.filter((change) => change.added)
                expect(jsx).toMatchSnapshot()
            } catch (error) {
                expect(error.message).toMatchSnapshot()
            }
        })
    })

    test('returns full AST when no previous data', async () => {
        const ast = await parseAndWaitForHighlighter({
            markdown,
            cache,
        })

        expect(toMarkdown(ast)).toMatchInlineSnapshot(`
          "# Welcome to the Example Document

          This is a sample markdown file to showcase various features.

          ## Features

          * Easy formatting
          * Lists
          * **Bold text**
          * [Links](https://example.com)

          > "Markdown makes documentation easy!"

          \`\`\`js
          console.log('Hello, world!');
          \`\`\`

          1. one
          2. two
          3. three

          this is another paragraph

          <Cards>
            <Card>
              some text
            </Card>
          </Cards>
          "
        `)
    })
})
