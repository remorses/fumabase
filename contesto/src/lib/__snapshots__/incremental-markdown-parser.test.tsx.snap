// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 1: "---" 1`] = `
"***
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 2: "--- title:" 1`] = `
"\\--- title:
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 3: "--- title: Test" 1`] = `
"\\--- title: Test
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 4: "--- title: Test Document" 1`] = `
"\\--- title: Test Document
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 5: "--- title: Test Document author:" 1`] = `
"\\--- title: Test Document author:
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 6: "--- title: Test Document author: John" 1`] = `
"\\--- title: Test Document author: John
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 7: "--- title: Test Document author: John Doe" 1`] = `
"\\--- title: Test Document author: John Doe
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 8: "--- title: Test Document author: John Doe ---" 1`] = `
"\\--- title: Test Document author: John Doe ---
"
`;

exports[`frontmatter-only file with newlines > frontmatter parsing at chunk 9: "--- title: Test Document author: John Doe --- " 1`] = `
"\\--- title: Test Document author: John Doe ---
"
`;

exports[`incremental parse > incremental parsing and jsx at chunk 1 1`] = `"<h1></h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 2 1`] = `"<h1>Welcome</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 3 1`] = `"<h1>Welcome to</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 4 1`] = `"<h1>Welcome to the</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 5 1`] = `"<h1>Welcome to the Example</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 6 1`] = `"<h1>Welcome to the Example Document</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 7 1`] = `"<h1>Welcome to the Example Document This</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 8 1`] = `"<h1>Welcome to the Example Document This is</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 9 1`] = `"<h1>Welcome to the Example Document This is a</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 10 1`] = `"<h1>Welcome to the Example Document This is a sample</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 11 1`] = `"<h1>Welcome to the Example Document This is a sample markdown</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 12 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 13 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 14 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 15 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 16 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features.</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 17 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features.</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 18 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 19 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features *</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 20 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 21 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 22 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting *</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 23 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 24 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists *</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 25 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * **Bold</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 26 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong></h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 27 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> *</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 28 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a></h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 29 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt;</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 30 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 31 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 32 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 33 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot;</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 34 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; \`\`\`js</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 35 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; \`\`\`js console.log(&#x27;Hello,</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 36 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; \`\`\`js console.log(&#x27;Hello, world!&#x27;);</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 37 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code></h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 38 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1.</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 39 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 40 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2.</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 41 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 42 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3.</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 43 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3. three</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 44 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3. three this</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 45 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3. three this is</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 46 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3. three this is another</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 47 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3. three this is another paragraph</h1>"`;

exports[`incremental parse > incremental parsing and jsx at chunk 48 1`] = `"Expected a closing tag for \`<Cards>\` (1:309-1:316) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing and jsx at chunk 49 1`] = `"Expected a closing tag for \`<Card>\` (1:317-1:323) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing and jsx at chunk 50 1`] = `"Expected a closing tag for \`<Card>\` (1:317-1:323) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing and jsx at chunk 51 1`] = `"Expected a closing tag for \`<Card>\` (1:317-1:323) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing and jsx at chunk 52 1`] = `"Expected a closing tag for \`<Cards>\` (1:309-1:316) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing and jsx at chunk 53 1`] = `"<h1>Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features * Easy formatting * Lists * <strong>Bold text</strong> * <a href="https://example.com" title="">Links</a> &gt; &quot;Markdown makes documentation easy!&quot; <code>js console.log(&#x27;Hello, world!&#x27;); </code> 1. one 2. two 3. three this is another paragraph <cards> <card> some text </card> </cards></h1>"`;

exports[`incremental parse > incremental parsing at chunk 1 1`] = `
"#
"
`;

exports[`incremental parse > incremental parsing at chunk 2 1`] = `
"# Welcome
"
`;

exports[`incremental parse > incremental parsing at chunk 3 1`] = `
"# Welcome to
"
`;

exports[`incremental parse > incremental parsing at chunk 4 1`] = `
"# Welcome to the
"
`;

exports[`incremental parse > incremental parsing at chunk 5 1`] = `
"# Welcome to the Example
"
`;

exports[`incremental parse > incremental parsing at chunk 6 1`] = `
"# Welcome to the Example Document
"
`;

exports[`incremental parse > incremental parsing at chunk 7 1`] = `
"# Welcome to the Example Document This
"
`;

exports[`incremental parse > incremental parsing at chunk 8 1`] = `
"# Welcome to the Example Document This is
"
`;

exports[`incremental parse > incremental parsing at chunk 9 1`] = `
"# Welcome to the Example Document This is a
"
`;

exports[`incremental parse > incremental parsing at chunk 10 1`] = `
"# Welcome to the Example Document This is a sample
"
`;

exports[`incremental parse > incremental parsing at chunk 11 1`] = `
"# Welcome to the Example Document This is a sample markdown
"
`;

exports[`incremental parse > incremental parsing at chunk 12 1`] = `
"# Welcome to the Example Document This is a sample markdown file
"
`;

exports[`incremental parse > incremental parsing at chunk 13 1`] = `
"# Welcome to the Example Document This is a sample markdown file to
"
`;

exports[`incremental parse > incremental parsing at chunk 14 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase
"
`;

exports[`incremental parse > incremental parsing at chunk 15 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various
"
`;

exports[`incremental parse > incremental parsing at chunk 16 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features.
"
`;

exports[`incremental parse > incremental parsing at chunk 17 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features.
"
`;

exports[`incremental parse > incremental parsing at chunk 18 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features
"
`;

exports[`incremental parse > incremental parsing at chunk 19 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\*
"
`;

exports[`incremental parse > incremental parsing at chunk 20 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy
"
`;

exports[`incremental parse > incremental parsing at chunk 21 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting
"
`;

exports[`incremental parse > incremental parsing at chunk 22 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\*
"
`;

exports[`incremental parse > incremental parsing at chunk 23 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists
"
`;

exports[`incremental parse > incremental parsing at chunk 24 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\*
"
`;

exports[`incremental parse > incremental parsing at chunk 25 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* \\*\\*Bold
"
`;

exports[`incremental parse > incremental parsing at chunk 26 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text**
"
`;

exports[`incremental parse > incremental parsing at chunk 27 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\*
"
`;

exports[`incremental parse > incremental parsing at chunk 28 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com)
"
`;

exports[`incremental parse > incremental parsing at chunk 29 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) >
"
`;

exports[`incremental parse > incremental parsing at chunk 30 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown
"
`;

exports[`incremental parse > incremental parsing at chunk 31 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes
"
`;

exports[`incremental parse > incremental parsing at chunk 32 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation
"
`;

exports[`incremental parse > incremental parsing at chunk 33 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!"
"
`;

exports[`incremental parse > incremental parsing at chunk 34 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \\\`\\\`\\\`js
"
`;

exports[`incremental parse > incremental parsing at chunk 35 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \\\`\\\`\\\`js console.log('Hello,
"
`;

exports[`incremental parse > incremental parsing at chunk 36 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \\\`\\\`\\\`js console.log('Hello, world!');
"
`;

exports[`incremental parse > incremental parsing at chunk 37 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \`
"
`;

exports[`incremental parse > incremental parsing at chunk 38 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1.
"
`;

exports[`incremental parse > incremental parsing at chunk 39 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one
"
`;

exports[`incremental parse > incremental parsing at chunk 40 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2.
"
`;

exports[`incremental parse > incremental parsing at chunk 41 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two
"
`;

exports[`incremental parse > incremental parsing at chunk 42 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3.
"
`;

exports[`incremental parse > incremental parsing at chunk 43 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3. three
"
`;

exports[`incremental parse > incremental parsing at chunk 44 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3. three this
"
`;

exports[`incremental parse > incremental parsing at chunk 45 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3. three this is
"
`;

exports[`incremental parse > incremental parsing at chunk 46 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3. three this is another
"
`;

exports[`incremental parse > incremental parsing at chunk 47 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3. three this is another paragraph
"
`;

exports[`incremental parse > incremental parsing at chunk 48 1`] = `"Expected a closing tag for \`<Cards>\` (1:309-1:316) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing at chunk 49 1`] = `"Expected a closing tag for \`<Card>\` (1:317-1:323) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing at chunk 50 1`] = `"Expected a closing tag for \`<Card>\` (1:317-1:323) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing at chunk 51 1`] = `"Expected a closing tag for \`<Card>\` (1:317-1:323) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing at chunk 52 1`] = `"Expected a closing tag for \`<Cards>\` (1:309-1:316) before the end of \`atxHeading\`"`;

exports[`incremental parse > incremental parsing at chunk 53 1`] = `
"# Welcome to the Example Document This is a sample markdown file to showcase various features. ## Features \\* Easy formatting \\* Lists \\* **Bold text** \\* [Links](https://example.com) > "Markdown makes documentation easy!" \`js console.log('Hello, world!'); \` 1. one 2. two 3. three this is another paragraph <Cards> <Card> some text </Card> </Cards>
"
`;
